contracts/REXTwoWayUniswapMarket.sol

```diff
@@ -0,0 +1,615 @@
+// SPDX-License-Identifier: AGPLv3
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./superswap/interfaces/ISwapRouter02.sol";
+
+import "./REXMarket.sol";
+import "./ISETHCustom.sol";
+
+contract REXTwoWayUniswapMarket is REXMarket {
+    using SafeERC20 for ERC20;
+
+    ISuperToken inputTokenA;
+    ISuperToken inputTokenB;
+    uint32 constant OUTPUTA_INDEX = 0;
+    uint32 constant OUTPUTB_INDEX = 1;
+    uint32 constant SUBSIDYA_INDEX = 2;
+    uint32 constant SUBSIDYB_INDEX = 3;
+    uint256 lastDistributionTokenAAt;
+    uint256 lastDistributionTokenBAt;
+    address public constant MATICX = 0x3aD736904E9e65189c3000c7DD2c8AC8bB7cD4e3;
+    ISuperToken subsidyToken;
+    ISwapRouter02 router =
+        ISwapRouter02(0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45);
```

Comment:
If there is room left in the contract, could you add methods to set `router`, `path`, and `poolFees` to an initialization method, like `initializedUniswap`, I have had a lot of errors hard coding these kinds of things here.